instrumentation/logging/logging.go

```diff
@@ -159,3 +115,67 @@ func (stdIO *stdIO) restore(file **os.File, replace bool) {
 		*file = stdIO.oldIO
 	}
 }
+
+// Handles the StdIO pipe for stdout and stderr
+func stdIOHandler(stdio *stdIO, isError bool) {
+	stdio.sync.Add(1)
```

Comment:
The `sync.Add` operation should be done in `Init` before starting the go routine, and also we should have a `stdio.sync.Wait()` at the end of that function, otherwise this does nothing.

Reply:
done